mri/operators/fourier/orc_wrapper.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+##########################################################################
+# pySAP - Copyright (C) CEA, 2017 - 2018
+# Distributed under the terms of the CeCILL-B license, as published by
+# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
+# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
+# for details.
+##########################################################################
+
+"""
+This modules contains classes for Off-Resonance Correction (ORC)
+
+:Author: Guillaume Daval-Frérot <guillaume.davalfrerot@cea.fr>
+"""
+
+import numpy as np
+
+from ..base import OperatorBase
+from .utils import (compute_mfi_coefficients,
+                    compute_mti_coefficients,
+                    compute_svd_coefficients,
+                    compute_fsvd_coefficients)
+
+
+class ORCFFTWrapper(OperatorBase):
+    """ Off-Resonance Correction FFT Wrapper
+
+    This class is used to wrap any FFT operator and change it
+    into an off-resonance correction multi-linear operator using
+    the method described in :cite: `sutton2003` with different
+    choices of coefficients described in :cite: `fessler2005`.
+
+    """
+    def __init__(self, fourier_op, field_map, time_vec, mask,
+                 coefficients="svd", weights="full", L="auto", n_bins="auto"):
+        """ Initialize and compute multi-linear correction coefficients
+
+        Parameters
+        ----------
+        fourier_op: OperatorBase
+            Cartesian or non-Cartesian Fourier operator to wrap
+        field_map: numpy.ndarray
+            B0 field inhomogeneity map (in Hz)
```

Comment:
I remember seeing that getting this map up was a painful task, do you think we should have this loader as a utils function here? Or shall we make it a part of `reconstruction_pipelines`?

Reply:
It depends deeply on how the field map is stored, so I don't think it should be part of pysap-mri.